lib/src/main/java/com/hootsuite/nachos/NachoTextView.java

```diff
@@ -727,6 +727,128 @@ public void setTextWithChips(@Nullable List<ChipInfo> chips) {
         endUnwatchedTextChange();
     }
 
+    /**
+     *
+     * @return String of unchipped text that AutoCompleteTextView sees
+     */
+
+    public String getTextForSuggestion() {
```

Comment:
⚠️   @tylerjroach  I'm not sure if this is what you intended, but I noticed something funny when using `getSelection()` and I will include a gif with it. Basically, if you use `getSelection` this method is at the mercy where the user's cursor is in the `NachoTextView`. Depending on how you want to create your suggestions with this bug and possibility of moving cursors in mind, I have some suggestions:

- Knowing where you  cursor position is, search for the closest chip end that comes before the current `token` (unchipped) and the closes chip start that comes after the `token` (unchipped) so you can provide the suggestion based on the text between chips.
- Keep what you have which is to find the closest chip end that comes before the current `token` (unchipped)` and return the text from that chip end to the cursor position (which may not be the end of the `token`

After thinking about this method and its purpose, I think of it as an application level functionality that is useful for your usage, but doesn't belong to what a `NachoTextView` is. At this moment, we want to keep the `NachoTextView` API simpler. We are suggesting to have this method either:

- A utility method on the NachoTextView (and extension method)
- A Custom View that has a NachoTextView and provides this extra method and other convenience methods.

If we are to include this method, it means we are formally including it into the public API of `NachoTextView` and continue to support it. We would rather keep the API lean and allow for other util methods to support `NachoTextView` instead of adding higher level usage methods such as this one.